GeneralsMD/Code/GameEngineDevice/Source/StdDevice/Common/StdLocalFileSystem.cpp

```diff
@@ -90,12 +90,16 @@ File * StdLocalFileSystem::openFile(const Char *filename, Int access /* = 0 */)
 			{
 				pathFixedPart = p;
 			}
+			else if (std::filesystem::exists(pathFixed / p, ec))
+			{
+				pathFixedPart = p;
+			}
 			else
 			{
 				// Check if the subpath exists using case-insensitive comparison
-				for (auto& entry : std::filesystem::directory_iterator(pathCurrent, ec))
+				for (auto& entry : std::filesystem::directory_iterator(pathFixed, ec))
```

Comment:
Using the fixed path to loop through, as otherwise it ends up not being correct inside the loop